mlflow/tracking/_tracking_service/client.py

```diff
@@ -80,7 +81,42 @@ def get_metric_history(self, run_id, key):
 
         :return: A list of :py:class:`mlflow.entities.Metric` entities if logged, else empty list
         """
-        return self.store.get_metric_history(run_id=run_id, metric_key=key)
+
+        # NB: Paginated query support is currently only available for the RestStore backend.
+        # FileStore and SQLAlchemy store do not provide support for paginated queries and will
+        # raise an MlflowException if the `max_results` argument is not None when calling this
+        # API.
+
+        if self.store == "databricks":
```

Comment:
If we scope the store to Databricks, then the AzureML folks and other backend store impls can't use the API via the MLflow client. Instead, can we update the store implementations so that they don't throw when `max_results` is specified (can still throw on `page_token`)? These store impls can ignore `max_results`. Then, can remove this `if` logic?

Reply:
sounds good!